### ISSUE TITLE
Can't open lib 'SQL Server Native Client 11.0' upgrading to 2.2.0 

### ISSUE THREAD

**Issue by flaviodelgrosso** [TimelordUK/node-sqlserver-v8#212]
Hi, I'm on macOS 11.4 ARM64 and noticed that upgrading to version 2.2.0 when I try to connect it returns the following error:

`ConnectionError: [unixODBC][Driver Manager]Can't open lib 'SQL Server Native Client 11.0' : file not found`

Using the 2.1.0 I've not this problem.
Anyone can suggest how to solve it?

Thanks in advance

**Comment by TimelordUK** [comment]
Can you try changing connection string so it is correctly named data source 

Driver={SQL Server Native Client 17.0}; Server=

We now communicate via ODBC manager so the driver must be exactly as installed

**Comment by flaviodelgrosso** [comment]
> Can you try changing connection string so it is correctly named data source Driver={ODBC Driver 17 for SQL Server} We now communicate via ODBC manager so the driver must be exactly as installed
> […](#)

Thanks for you reply! @TimelordUK 

I tried to use actually this:

```
const pool = new ConnectionPool({
        driver: 'ODBC Driver 17 for SQL Server',
        server: host,
        user,
        password,
        port: parseInt(port, 10),
        options: {
          enableArithAbort: true,
          trustedConnection: winauth === 'Y',
          trustServerCertificate: true,
        },
      });
```

If the driver key/value is syntactically correct, it's not working for me..

**Comment by TimelordUK** [comment]
are you using node-mssql?  if you are i think it places the Driver={SQL
Server Native Client 11.0} in the connection string - this  will no longer
work.

you can try importing the library directly in a test program such as

```js
const query = 'SELECT * FROM syscomments'

sql.open(connectionString, function (err, con) {
  if (err) {
    console.log(`failed to open ${err.message}`)
  }
  let d = new Date()
  con.query(query, function (err, rows) {
```

**Comment by flaviodelgrosso** [comment]
> are you using node-mssql? if you are i think it places the Driver={SQL Server Native Client 11.0} in the connection string - this will no longer work. you can try importing the library directly in a test program such as const query = 'SELECT * FROM syscomments' sql.open(connectionString, function (err, con) { if (err) { console.log(`failed to open ${err.message}`) } let d = new Date() con.query(query, function (err, rows) { On Mon, 7 Jun 2021 at 09:31, flaviodelgrosso ***@***.***> wrote:

Yes I'm importing node-mssql as const { ConnectionPool } = require('mssql/msnodesqlv8');

So the problem is related to this actually. Thanks for your explanation

**Comment by TimelordUK** [comment]
![image](https://user-images.githubusercontent.com/7300251/121150839-c6457b00-c83b-11eb-8bc5-abe9cb9ee604.png)


you could try configuring a new source named the missing driver as above but pointing to odbc17 - I have not tried this but should work ok. ODBC manager simply needs to map the name given in driver to the shared library to load.  It needs to know how to map the 'SQL Server Native Client 11.0' which could be added into data sources

**Comment by flaviodelgrosso** [comment]
> you could try configuring a new source named the missing driver as above but pointing to odbc17 - I have not tried this but should work ok. ODBC manager simply needs to map the name given in driver to the shared library to load. It needs to know how to map the 'SQL Server Native Client 11.0' which could be added into data sources

Yes that's working, thank you!

**Comment by SixBytesUnder** [comment]
@flaviodelgrosso I'm having the same issue, can you share how you fixed it? Everything worked just fine when I was on a version `2.0.5`. few days ago I updated to `2.4.0` and suddenly I'm getting `ConnectionError: [unixODBC][Driver Manager]Can't open lib 'SQL Server Native Client 11.0' : file not found`.
I'm on CentOS 8 using Node v12.18.3

How do you configure a new source and what do I enter as value for "driver"?

      const config = {
        user: process.env.AX_USERNAME,
        password: process.env.AX_PASSWORD,
        server: process.env.AX_SERVER,
        database: process.env.AX_DATABASE,
        driver: 'msnodesqlv8'
      };
      sql.connect(config).then((pool) => {
        ...
      });

Above worked fine in 2.0.5, but fails in 2.4.0

    $ odbcinst -j
    
    unixODBC 2.3.7
    DRIVERS............: /etc/odbcinst.ini
    SYSTEM DATA SOURCES: /etc/odbc.ini
    FILE DATA SOURCES..: /etc/ODBCDataSources
    USER DATA SOURCES..: /home/ecom/.odbc.ini
    SQLULEN Size.......: 8
    SQLLEN Size........: 8
    SQLSETPOSIROW Size.: 8

    $ cat /etc/odbcinst.ini

    [ODBC Driver 17 for SQL Server]
    Description=Microsoft ODBC Driver 17 for SQL Server
    Driver=/opt/microsoft/msodbcsql17/lib64/libmsodbcsql-17.8.so.1.2
    UsageCount=1

**Comment by TimelordUK** [comment]
thnaks for update I will add to the Wiki - preveiously msnodesqlv8 linked directly to the MS driver, now we link to ODBC which manages all communication to the database, hence ODBC does need this entry mapping.  Exactly as you worked out above.

**Comment by TimelordUK** [comment]
something along these lines 

```sh
cd /usr/lib
sudo rm libssl.so.1.1
export LD_PRELOAD=/home/linuxbrew/.linuxbrew/opt/openssl@1.1/lib/libssl.so.1.1
sudo ln -s /home/linuxbrew/.linuxbrew/opt/openssl@3/lib/libssl.so  libssl.so.1.1
ls -l /lib/x86_64-linux-gnu/libssl.so.1.1
cd /lib/x86_64-linux-gnu/
ls -alrt
sudo rm libssl.so.1.1
sudo ln -s /home/linuxbrew/.linuxbrew/opt/openssl@3/lib/libssl.so  libssl.so.1.1
ls -l /lib/x86_64-linux-gnu/libssl.so.1.1

```

**Comment by TimelordUK** [comment]
im closing this issue the binaries for apple silicon, ubuntu and nwindows are now generated by git actions
